src/command.coffee

```diff
@@ -102,7 +102,17 @@ exports.run = ->
   process.argv = process.argv[0..1].concat literals
   process.argv[0] = 'coffee'
 
-  opts.output = path.resolve opts.output  if opts.output
+  if opts.output
+    outputBasename = path.basename opts.output
+    if '.' in outputBasename
```

Comment:
Okay, so now if `path.basename` (which is everything from the last separator, `/` or `\`, to the end of the string) returns only periods, I’m just treating the output argument as a path (folder). This conforms to the old behavior, though it doesn’t seem to make much sense.

If the output argument ends in a slash, I’m forcing it to be treated as a folder name, even if it contains periods. So `coffee -co scripts.js/ test.coffee` will save into `scripts.js/test.js`.